File: dart/difference-of-squares/lib/difference_of_squares.dart
class DifferenceOfSquares {
  int squareOfSum(int n) {
    // n(n + 1)/2
    var sum = ((n * (n + 1)) ~/ 2);
    return sum * sum;
  }

  int sumOfSquare(int n) {
    // n(n + 1)(2n + 1)/6
    return ((n * (n + 1) * ((2 * n) + 1)) ~/ 6);
  }

  int differenceOfSquares(int n) {
    return squareOfSum(n) - sumOfSquare(n);
  }
}

File: swift/difference-of-squares/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
public struct Squares {
  var squareOfSum: Int
  var sumOfSquares: Int
  var differenceOfSquares: Int

  init(_ n: Int) {
    let sum = n * (n + 1) / 2
    squareOfSum = sum * sum
    sumOfSquares = n * (n + 1) * ((2 * n) + 1) / 6
    differenceOfSquares = squareOfSum - sumOfSquares
  }
}